File: Design-Patterns-Using-Kotlin/Basics OOPS/AssociationRelationship.kt
open class Student( val name: String) {
    override fun toString(): String {
        return "Student: $name"
    }
}

class SchoolClass(private val className: String) {
    private val students: MutableList<Student> = ArrayList()

    fun addStudent(student: Student) {
        students.add(student)
    }

    fun display() {
        for (student in students) {
            println(student)
        }
    }
}

fun main() {
    val schoolClass = SchoolClass("SS1")
    val student1 = Student("John Smith")
    val student2 = Student("Jane Smith")

    schoolClass.addStudent(student1)
    schoolClass.addStudent(student2)

    schoolClass.display()
}

/*
Student: John Smith
Student: Jane Smith
*/

File: Design-Patterns-Using-Swift/Basics OOPS/AssociationRelationship.swift
class Student {
    private var name: String

    init(name: String) {
        self.name = name
    }

    var description: String {
        return "Student: \(name)"
    }
}

class SchoolClass {
    private var className: String
    private var students: [Student]

    init(className: String) {
        self.className = className
        self.students = []
    }

    func addStudent(_ student: Student) {
        students.append(student)
    }

    func display() {
        for student in students {
            print(student.description)
        }
    }
}

let schoolClass = SchoolClass(className: "SS1")
let student1 = Student(name: "John Smith")
let student2 = Student(name: "Jane Smith")

schoolClass.addStudent(student1)
schoolClass.addStudent(student2)

schoolClass.display()
